### DISCUSSION TITLE
Support `export const page` as well as `export default`

### DISCUSSION THREAD

**Discussion by moshest** [vercel/next.js#38966] [Ideas]
### Describe the feature you'd like to request

Our current code base deprecate the use of `export default` in favor of named export. We are using eslint `import/no-default-export` role to enforce it and currently our pages file looks like this:

```ts
// eslint-disable-next-line import/no-default-export
export default Page;
```

Read here more about what are the reasons to deprecate default export:
https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad

Related: https://github.com/vercel/next.js/issues/8214

### Describe the solution you'd like

It would be nice if Next.js could support also named export `page` for pages, for example:

```ts
export const page = Page;
```

### Describe alternatives you've considered

For safety and backward compatibility, Next.js should always check first for default export and only then for the name export. The name export can be a fixed name like `page` or just search for any exported component and use it if only one found (throw if multiple found).

For example:

```ts
export const MyPage: FC = () => {
   return <div>Welcome to Next.js!</div>;
};
```

**Comment by bryanprimus** [comment]
not sure if this is possible. maybe we can disable that eslint rules to only apply to every dir but pages? so we don't need to put eslint disable in every pages file
